Autopilot/PathManager/Src/landingTakeoffManager.cpp

```diff
@@ -0,0 +1,202 @@
+#include "landingTakeoffManager.hpp"
+#include <math.h>
+
+double LandingManager::changingAltitude(const SFOutput_t * input, const _PathData * aimingPoint, const _PathData * intersectionPoint, const _PathData * stoppingPoint)
+{
+    //vector declaration
+    Vector3D aPoint(aimingPoint->longitude, aimingPoint->latitude, aimingPoint->altitude); //aiming point
+    Vector3D iPoint(intersectionPoint->longitude, intersectionPoint->latitude, intersectionPoint->altitude); //intersection point
+    Vector3D sPoint(stoppingPoint->longitude, stoppingPoint->latitude, stoppingPoint->altitude); //stopping point
+    Vector3D cPoint(input->longitude, input->latitude, input->altitude); //current point
+    Vector3D projectedPoint;
+
+    //calculating the vectors
+    Vector3D vectorAI(iPoint.x - aPoint.x, iPoint.y - aPoint.y, iPoint.z - aPoint.z); //aiming point --> intersection point
+    Vector3D vectorAS(sPoint.x - aPoint.x, sPoint.y - aPoint.y, sPoint.z - aPoint.z); //aiming point --> stopping point
+
+    Vector3D vectorAC(cPoint.x - aPoint.x, cPoint.y - aPoint.y, cPoint.z - aPoint.z); //aiming point --> current point
+    Vector3D normalVector;
+    
+    //normal of the plane created by aiming, stopping, and intersection point
+    normalVector = vectorAI.crossProduct(vectorAS);
+
+    //projecting vectorAC to the normal
+    //equation is the dot product of normal and vectorAC, divided by the norm of the normal squared, multiplied by the normal vector
+    Vector3D projectionACToNormal;
+
+    projectionACToNormal = normalVector * (normalVector.dotProduct(vectorAC) / pow(normalVector.norm(),2));
+
+    //project point onto plane (cpoint minus the projection)
+
+    projectedPoint = cPoint - projectionACToNormal;
+
+    //determine the equation of the line of the slope and sub in projected values (symmetric form)
+    double altitude = (((projectedPoint.x - aPoint.x) / vectorAI.x) + aPoint.z) * vectorAI.z;
+
+    return altitude;
+}
+
+double LandingManager::approachSpeed(const bool *ifPackage)
```

Comment:
noted before, but emphasizing here. Reason to pass by reference is to save copying space/time. THat's good for custom structs, but since a bool and a pointer to a bool take up the same space, there's no need to pass a bool by reference.